pages/studentInstanceQuestionHomework/studentInstanceQuestionHomework.js

```diff
@@ -103,10 +103,10 @@ router.post('/', function(req, res, next) {
 
 router.get('/', function(req, res, next) {
     if (res.locals.assessment.type !== 'Homework') return next();
-    // req.query.variant_id might be undefined, which will generate a new variant
-    question.getAndRenderVariant(req.query.variant_id, res.locals, function(err) {
+    logPageView(req, res, (err) => {
         if (ERR(err, next)) return;
-        logPageView(req, res, (err) => {
+        // req.query.variant_id might be undefined, which will generate a new variant
+        question.getAndRenderVariant(req.query.variant_id, res.locals, function(err) {
```

Comment:
As above, we need `logPageView()` to come after `getAndRenderVariant()` so that it can access the newly set `res.locals.variant`.